emmet-builders/emmet/builders/mobility_screening/migration_graph.py

```diff
@@ -0,0 +1,66 @@
+class MigrationGraphBuilder(MapBuilder):
+    def __init__(
+        self,
+        electrodes: MongoStore,
+        tasks: MongoStore,
+        migration_graphs: MongoStore,
+        ltol: float = 0.4,
+        stol: float = 0.6,
+        angle_tol: float = 15,
+        **kwargs
+    ) -> MapBuilder:
+        self.electrodes = electrodes
+        self.tasks = tasks
+        self.ltol = ltol
+        self.stol = stol
+        self.angle_tol = angle_tol
+        self.migration_graphs = migration_graphs
+        self.wi_entries = {}
+
+        super().__init__(source=electrodes, target=migration_graphs, **kwargs)
+        self.sources.append(tasks)
+
+    def get_items(self) -> dict:
+        for item in super(MigrationGraphBuilder,self).get_items():
+            tds = list(self.tasks.query({"task_id":{"$in":item["material_ids"]}}))
+            item.update({"task_docs":tds})
+            yield item
+
+    #get structure entry from task store and attach migration graph dict to item
+    def unary_function(self, item: dict) -> dict:
+        new_item = dict(item)
+        task_documents = [TaskDocument.parse_obj(td) for td in new_item["task_docs"]]
+        entries = [task_doc.structure_entry for task_doc in task_documents]
+        #get entry is working_ion is new. limit MPRester calls
+        if new_item["working_ion"] not in self.wi_entries.keys():
+            self.wi_entries[new_item["working_ion"]] = min(compatibility.process_entries(
+                mpr.get_entries_in_chemsys([new_item["working_ion"]])),
+                                                           key = lambda x : x.energy_per_atom)
+        mg = self.get_migration_graph(compatibility.process_entries(entries),
+                                 self.wi_entries[new_item["working_ion"]],
+                                 ltol=self.ltol, stol=self.stol, angle_tol=self.angle_tol)
+        new_item["mg"] = mg
+        return new_item
+
+    def get_migration_graph(self, cses: [ComputedEntry], wi_entry: [ComputedEntry], max_distance_min=0.1, max_distance_max=10, ltol=0.4, stol=0.6, angle_tol=15) -> dict:
```

Comment:
I could be wrong, but I believe the type for cses should be ComputedStructureEntry? If you only provide computed entries without structures, I don't think MigrationGraph.get_structure_from_entries() will work. Also if cse is short for ComputerStructureEntry, you will want to be consistent with your variable names to reduce confusion.